File: tlias-frontend/src/api/student.ts
import request, { type PageResult } from './request'

export interface Student {
  id?: number
  name: string
  no: string
  gender: number
  phone: string
  idCard: string
  isCollege: number
  address: string
  degree: number
  graduationDate: string
  violationCount?: number
  violationScore?: number
  clazzId: number
  clazzName?: string
  createTime?: string
  updateTime?: string
}

export interface StudentQueryParam {
  page: number
  pageSize: number
  name?: string
  degree?: number
  clazzId?: number
}

// 学员分页条件查询
export function getStudentPage(params: StudentQueryParam) {
  return request.get<PageResult<Student>, PageResult<Student>>('/students', {
    params,
  })
}

// 根据 ID 查询学员
export function getStudentById(id: number) {
  return request.get<Student, Student>(`/students/${id}`)
}

// 新增学员
export function addStudent(data: Student) {
  return request.post('/students', data)
}

// 修改学员
export function updateStudent(data: Student) {
  return request.put('/students', data)
}

// 批量删除学员（请求体 JSON 数组，对应后端 @DeleteMapping + @RequestBody List<Integer>）
export function deleteStudent(ids: number[]) {
  return request.delete('/students', { data: ids })
}

// 违纪处理（扣分）
export function handleViolation(id: number, score: number) {
  return request.put(`/students/violation/${id}/${score}`)
}

// 学历映射
export const degreeMap: Record<number, string> = {
  1: '初中',
  2: '高中',
  3: '大专',
  4: '本科',
  5: '硕士',
  6: '博士',
}

File: tlias-web-management/src/main/java/com/kk/pojo/Student.java
package com.kk.pojo;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class Student {
    private Integer id;
    private String name;
    private String no; //学号
    private Integer gender; //1男 2女
    private String phone;
    private String idCard; //身份证
    private String address;
    private Integer degree; //学历
    private Integer clazzId; //所属班级id

    private Integer violationCount; //违纪次数
    private Integer violationScore; //违纪扣分

    private LocalDateTime createTime;
    private LocalDateTime updateTime;

    //=======【扩展：用于页面回显，数据库没有此字段！！】=======
    //页面表格要展示班级名称，数据库student表只有clazz_id，没有班级名
    private String clazzName;
}
